segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/ZKSegmentContainerMonitor.java

```diff
@@ -106,9 +107,16 @@ public void initialize() {
     public void initialize(Duration monitorInterval) {
         Exceptions.checkNotClosed(closed.get(), this);
 
+        // Start loading the segment container to node assigment map from zookeeper.
         this.hostContainerMapNode.start();
+
+        // There are two triggers for the segment container monitor.
+        // 1. On any segment container ownership changes notified via zookeeper. We will ensure the local containers
+        //      are stopped/started according to the new ownership mapping.
+        // 2. At scheduled intervals to perform retries on local segment container start failures.
         this.assigmentTask.set(this.executor.scheduleWithFixedDelay(
```

Comment:
Yes, this is redundant in the sense that most of the time it does nothing (an inmemory check and exits). A more efficient way (to avoid the regular checks) would be to schedule the task conditionally only at the end of start/stop container tasks on success or failure (via async functions), but complicates the code a little; will need to check and selectively schedule checkAssignment only for specific scenarios. The checkAssignment is just an inmemory check without any I/O operations when it doesn't have any tasks to perform. Using the fixed scheduler here makes the implementation very simple.